src/App.tsx

```diff
@@ -1,8 +1,113 @@
-import React from 'react';
+import React, { useState } from 'react';
+import TodosList from './TodosList';
 import './App.css';
+import { todosFromServer } from './API';
 
-const App = () => (
-  <h1>Dynamic list of TODOs</h1>
-);
+const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>([]);
+  const [isLoading, setLoading] = useState<boolean>(false);
+  const [isFirstLetterAlphabettName, setUpAlphabetName] = useState<boolean>(false);
+  const [isFirstLetterAlphabetTitle, setUpAlphabetTitle] = useState<boolean>(false);
+  const [isCompletedTodo, setIsCompletedTodo] = useState<boolean>(false)
+
+  const fetchData = (): void => {
+    setLoading(true);
+    todosFromServer.then(data => {
+      setLoading(false);
+
+      return setTodos(data);
+    });
+  };
+
+  const nameFilter = () => {
+    let newTodos = [];
+
+    if (!isFirstLetterAlphabettName) {
+      newTodos = [...todos].sort((a, b) => a.user.name.localeCompare(b.user.name));
+      setUpAlphabetName(true);
+
+      return setTodos(newTodos);
+    }
+
+    newTodos = [...todos].sort((a, b) => b.user.name.localeCompare(a.user.name));
+    setUpAlphabetName(false);
+
+    return setTodos(newTodos);
+  };
+
+  const titleFilter = () => {
+    let newTodos = [];
+
+    if (!isFirstLetterAlphabetTitle) {
+      newTodos = [...todos].sort((a, b) => a.title.localeCompare(b.title));
+      setUpAlphabetTitle(true);
+
+      return setTodos(newTodos);
+    }
+
+    newTodos = [...todos].sort((a, b) => b.title.localeCompare(a.title));
+    setUpAlphabetTitle(false);
+
+    return setTodos(newTodos);
+  };
+
+  const completedFilter = () => {
+    let newTodos = [];
+
+    if (!isCompletedTodo) {
+      newTodos = todos.sort((a, b) => Number(a.completed) - Number(b.completed));
+      setIsCompletedTodo(true);
+
+      return setTodos([...newTodos]);
+    }
+
+    newTodos = todos.sort((a, b) => Number(b.completed) - Number(a.completed));
+    setIsCompletedTodo(false);
+
+    return setTodos([...newTodos]);
```

Comment:
you still modify the initial todos, you need to store only initial todos in the state, all modified arrays of todos should be calculated directly before rendering